public/app/features/logs/utils.ts

```diff
@@ -220,18 +220,22 @@ export const mergeLogsVolumeDataFrames = (dataFrames: DataFrame[]): { dataFrames
 
     configs[level] = {
       meta: dataFrame.meta,
-      valueFieldConfig: valueField.config,
-      timeFieldConfig: timeField.config,
+      valueFieldConfig: valueField?.config ?? {},
+      timeFieldConfig: timeField?.config ?? {},
     };
 
-    for (let pointIndex = 0; pointIndex < length; pointIndex++) {
-      const time: number = timeField.values[pointIndex];
-      const value: number = valueField.values[pointIndex];
-      aggregated[level] ??= {};
-      aggregated[level][time] = (aggregated[level][time] || 0) + value;
-
-      totals[time] = (totals[time] || 0) + value;
-      maximumValue = Math.max(totals[time], maximumValue);
+    if (timeField && valueField) {
```

Comment:
Just a nit, but I'm a fan of early returns. Can't we just  return before doing `configs[level] = {` etc. if `timeField` or `valueField` is not set?

Reply:
Yeah, better. I also don't like deep nesting.

Reply:
Still wondering if we need to set `configs[level]` if those fields is not set. I think we should just skip that too.

Reply:
With other words, move the early return before

```
configs[level] = {
      meta: dataFrame.meta,
      valueFieldConfig: valueField?.config ?? {},
      timeFieldConfig: timeField?.config ?? {},
    };
 ```
 
 to line 220

Reply:
Moved.